wfc_core/src/lib.rs

```diff
@@ -299,27 +306,22 @@ impl World {
                 continue;
             }
 
-            let entropy = if self.use_shannon_entropy {
-                if let Some(entropy) = self.slot_entropy_lru.get(slot) {
-                    *entropy
-                } else {
-                    let mut sum_weights = 0.0;
-                    let mut sum_weight_log_weights = 0.0;
-                    for module in slot {
-                        let weight = self.module_weights[usize::from(module)];
-                        sum_weights += weight;
-                        sum_weight_log_weights += weight * weight.ln();
-                    }
-
-                    debug_assert!(sum_weights >= 0.0);
+            let entropy = if self.slot_module_weights_customized {
+                let mut sum_weights = 0.0;
+                let mut sum_weight_log_weights = 0.0;
+                for module in slot {
+                    let weight_index = self.module_count * i + usize::from(module);
+                    let weight = self.slot_module_weights[weight_index];
+                    sum_weights += weight;
+                    sum_weight_log_weights += weight * weight.ln();
+                }
```

Comment:
We need validation for invalid weights (negative, zero, infinity, NaN). Otherwise we get NaNs here.